ashlar/reg.py

```diff
@@ -827,7 +855,7 @@ def _calculate(self, corners1, corners2, min_size):
         max_shape = (corners2 - corners1).max(axis=0)
         min_size = min_size.clip(1, max_shape)
         position = corners1.max(axis=0)
-        initial_shape = np.ceil(corners2.min(axis=0) - position).astype(int)
+        initial_shape = np.floor(corners2.min(axis=0) - position).astype(int)
```

Comment:
I ran into the bug here just today when testing with synthetic data. One of the values in here was X.0000000001 so the `ceil` took it up to the next integer which was wrong and caused an inconsistency in the intersection shape vs. the offsets. I'm not so sure `floor` won't cause the same problem in the other direction... Either way, I think the refactor branch code for this is far easier to follow and reason about, so hopefully this code won't live for much longer.

Reply:
I did also run into the bug with `np.floor` some time ago, but it appears to me that it was less frequent compared to `np.ceil`. Agree on the refactor branch and we should figure out how to fix this problem!